apis/python/tests/test_dataframe_index_columns.py

```diff
@@ -1346,69 +1592,308 @@ def arrow_table():
         [
             "tsns-pa-array-untyped",
             ["tsns"],
+            None,
             [pa.array([946684800000000002, 946684800000000003])],
             "default23",
         ],
         [
             "tsns-pa-array-typed",
             ["tsns"],
+            None,
             [pa.array([946684800000000002, 946684800000000003], pa.timestamp("ns"))],
             "default23",
         ],
-        # Index by bool -- not currently supported
+    ],
+)
+def test_types_no_errors(
+    tmp_path,
+    arrow_table,
+    name,
+    index_column_names,
+    domain,
+    coords,
+    expecteds,
+):
+    uri = tmp_path.as_posix()
+
+    soma.DataFrame.create(
+        uri,
+        schema=arrow_table.schema,
+        index_column_names=index_column_names,
+        domain=domain,
+    )
+
+    with soma.DataFrame.open(uri, "w") as sdf:
+        sdf.write(arrow_table)
+
+    if expecteds == "default01234":
+        expecteds = {
+            "soma_joinid": pa.array([0, 1, 2, 3, 4], pa.int64()),
+            "string": pa.array(
+                ["apple", "ball", "cat", "dog", "egg"], pa.large_string()
+            ),
+        }
+
+    elif expecteds == "default23":
+        expecteds = {
+            "soma_joinid": pa.array([2, 3], pa.int64()),
+            "string": pa.array(["cat", "dog"], pa.large_string()),
+        }
+
+    with soma.DataFrame.open(uri, "r") as sdf:
+        actual_table = sdf.read(coords=coords).concat()
+        for query_column_name, expected_array in expecteds.items():
+            actual_array = actual_table[query_column_name].combine_chunks()
+            # The output from the first one is easier to read when it fails
+            assert actual_array.to_pylist() == expected_array.to_pylist()
+            assert actual_array == expected_array
+        if domain is not None:
+            actual_domain = sdf.domain
+            for i in range(len(domain)):
+                if (
```

Comment:
can we add a test that verifies what `domain` returns when the index column type is `string` or `bytes`?